### PR TITLE
[Fix](Nereids) fix floor/round/ceil/truncate functions type compute precision problem

### LINKED ISSUE #43421: [Bug] floor function execution failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master/2.1.7 

### What's Wrong?

when running 
with cte as (select floor(300.343, 2) order by 1 limit 1) select * from cte;
it rise a exception

### What You Expected?

sql could be executed correctly

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
### What problem does this PR solve?
- Problem
function like ```select floor(300.343, 2)``` precision should be 5 and scale should be 2, but now is (6, 2) after compute precision, but after folding const on fe, it changed to (5, 2) but upper level of plan still expect the output of child to be (6, 2). So it would rise an exception when executing. 
- How it was fixed 
fix folding constant precision of floor/round/ceil/truncate functions from (5, 2) to (6, 2) in upper case
- Notion
when second value is negative and it absolute value >= precision - value, it can not be expressed in fe which result is zero with decimal type (3, 0). like 000. So just let it go back and no using folding constant by fe.

Issue Number: close #43421

Related PR: #40744

Problem Summary:

### Check List (For Author)

- Test 

    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [x] No.
    - [ ] Yes. 

- Does this need documentation?

    - [x] No.
    - [ ] Yes. 

- Release note

    
    
    Fix floor/round/ceil functions precision problem

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: regression-test/suites/nereids_p0/expression/fold_constant/fold_constant_numeric_arithmatic.groovy [hunk 1]
```diff
@@ -409,4 +409,13 @@ test {
...
     testFoldConst("SELECT CONV(-10, 10, 2) AS conv_invalid_base") //Conv with negative input (may be undefined)
+
+    // fix floor/ceil/round function return type with DecimalV3 input
+    testFoldConst("with cte as (select floor(300.343) order by 1 limit 1) select * from cte")
+    testFoldConst("with cte as (select round(300.343) order by 1 limit 1) select * from cte")
+    testFoldConst("with cte as (select ceil(300.343) order by 1 limit 1) select * from cte")
+
+    testFoldConst("with cte as (select floor(300.343, 2) order by 1 limit 1) select * from cte")
+    testFoldConst("with cte as (select round(300.343, 2) order by 1 limit 1) select * from cte")
+    testFoldConst("with cte as (select ceil(300.343, 2) order by 1 limit 1) select * from cte")
```
Comment (superdiaodiao) [inline regression-test/suites/nereids_p0/expression/fold_constant/fold_constant_numeric_arithmatic.groovy:420]: I tested the query before this PR:
`with cte as (select truncate(300.343, 2) order by 1 limit 1) select * from cte;`, 
and the exception is:
`ERROR 1105 (HY000): errCode = 2, detailMessage = (9.134.238.135)[INTERNAL_ERROR]output type not match expr type  , col name  , expected type Decimal(6, 2) , real type Decimal(5, 2)`.
So whether the truncate() should be tested together?

Comment (LiBinfeng-01) [reply regression-test/suites/nereids_p0/expression/fold_constant/fold_constant_numeric_arithmatic.groovy:420]: it make sense, added

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run p0

**Comment by LiBinfeng-01** [comment]
run cloud_p0

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java [hunk 2]
```diff
@@ -1136,21 +1136,10 @@ public static Expression mathE() {
     public static Expression truncate(DecimalV3Literal first, IntegerLiteral second) {
         if (first.getValue().compareTo(BigDecimal.ZERO) == 0) {
             return first;
+        } else if (first.getValue().compareTo(BigDecimal.ZERO) == -1) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java:1139]:
```suggestion
        } else if (first.getValue().compareTo(BigDecimal.ZERO) < 0) {
```

Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java:1139]: the function compareTo only return 0 -1 and 1, so I think -1 is acceptable for this situation

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java (hunk 2 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java:1144]: you wrong. https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html#compareTo-T-

Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java:1144]: https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html
maybe you can check bigDecimal Type implementation for compareTo

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java (hunk 2 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java]: changed to aligning interface definition

**Review by englefly — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**